### PR TITLE
chore: update blocks to use primitive cubes

### LINKED ISSUE #40: Render cube instead of 6 quads
Current implementation of cube is using 6 quads and this results in high draw count. 
Doing a comparison with 192 cubes.

Current block:
image

Cubes:
image

As shown there is a significant fps increase due to reduce in draw count, if we were to go with these cubes. We need to be able to change how block face hovering and clicking will work.

### PR DESCRIPTION
**Pull request by nicholaschuayunzhi**
fixes #40 
Previously we were using 6 quads resulting in high draw count.

# Changes
Now face does not highlight on hover/click. Will have to find a work around.
Introduces a `PlayerMouse` `Monobehaviour`

image

1000 Blocks:
image

### CODE REVIEW

File: Assets/Scripts/BlockFaceBehaviour.cs [hunk 1]
```diff
@@ -6,116 +6,57 @@
...
-        // quad vertices are in this order:
-        // 3 -C- 1 
-        // B     D with the normal facing you
-        // 0 -A- 2
-        // 
-        Vector3 a20 = worldCoord[0] - worldCoord[2];
-        Vector3 b03 = worldCoord[3] - worldCoord[0];
-        Vector3 c31 = worldCoord[1] - worldCoord[3];
-        Vector3 d12 = worldCoord[2] - worldCoord[1];
+        float centerToSkin = FaceLength * (1 - skinToLengthRatio / 2) / 2;
```
Comment (Rinder5) [inline Assets/Scripts/BlockFaceBehaviour.cs]: ~is this supposed to be a ratio or offset? Ratios' value tend to rely on the original value, while offsets should stay constant either way. The way this is written seems to imply it's an offset~

File: Assets/Scripts/BlockFaceBehaviour.cs [hunk 2]
```diff
@@ -6,116 +6,57 @@
...
         Vector3[] skinVertices = new Vector3[4];
-        skinVertices[0] = worldCoord[0] - (a20 * skinWidth) + (b03 * skinWidth) - (_normal * skinWidth);
-        skinVertices[1] = worldCoord[1] - (c31 * skinWidth) + (d12 * skinWidth) - (_normal * skinWidth);
-        skinVertices[2] = worldCoord[2] + (a20 * skinWidth) + (b03 * skinWidth) - (_normal * skinWidth);
-        skinVertices[3] = worldCoord[3] + (c31 * skinWidth) + (d12 * skinWidth) - (_normal * skinWidth);
+
+        skinVertices[0] = quadCenter + (centerToSkin * perpendicularNormals[0]) + (centerToSkin * perpendicularNormals[1]);
+        skinVertices[1] = quadCenter + (centerToSkin * perpendicularNormals[0]) - (centerToSkin * perpendicularNormals[1]);
+        skinVertices[2] = quadCenter - (centerToSkin * perpendicularNormals[0]) + (centerToSkin * perpendicularNormals[1]);
+        skinVertices[3] = quadCenter - (centerToSkin * perpendicularNormals[0]) - (centerToSkin * perpendicularNormals[1]);
```
Comment (Rinder5) [inline Assets/Scripts/BlockFaceBehaviour.cs]: Is it possible to use a variable to reduce the amount of similar code here?

**Review by Rinder5 — COMMENTED** [review]
Thanks for your hard work! Just a few minor changes which could be addressed before merging. Keep it up! =)

File: Assets/Scripts/BlockFaceBehaviour.cs (hunk 1 above)
Comment (Rinder5) [reply Assets/Scripts/BlockFaceBehaviour.cs]: actually, this is probably right. my bad! >.<

**Comment by nicholaschuayunzhi** [comment]
@Rinder5 have made changes requested.

**Comment by nicholaschuayunzhi** [comment]
@Rinder5, request for review. Want to get this merge so that the rest of the features don't get blocked.

**Pull request merged**